### PR TITLE
Stop publishing lakeFS DuckDB image

### LINKED ISSUE #6001: Remove DuckDB from lakeFS Docker image
## Why

We recently started maintaining a DuckDB binary as part of a set of lakeFS images `lakefs:...-with-duckdb`.  This makes no sense:

* Images are the basis for service oriented architectures, especially microservices.  Packaging DuckDB and lakeFS in the same image makes no sense.
* It is unreasonable to expect DuckDB to support users who use some unknown version of DuckDB that they found on a Docker image treeverse/lakefs.
* We cannot protect our users from DuckDB bugs because it's not our code.  Now we need to rebuild for any DuckDB bugs.  This includes fixes for security vulnerabilities -- for which we will have to work urgently in order to deliver a late fix.
* We should ensure compliance with [the DuckDB license](https://github.com/duckdb/duckdb/blob/master/LICENSE), and need to keep doing so for every version.  (Right now we may have to add a notice to our image.)

I believe the best way our is to remove DuckDB from lakeFS Docker images.

## Alternatives

If we decide we do need DuckDB, there are multiple options.  A few easy ones:

* Use WASM DuckDB
* Use the [DuckDB Homebrew formula](https://formulae.brew.sh/formula/duckdb), Linux users have [supported installation options](https://duckdb.org/docs/installation/index).
* Ask DuckDB to produce an image, and/or send them a Dockerfile and ask them to build an authorized version.
* Provide a separate treeverse/duckdb image.  This resolves _some_ but not _all_ of the issues.
* Provide our users with a Dockerfile they can use to build it.

### PR DESCRIPTION
**Pull request by rmoff**
### Linked Issue

Closes #6001

---

### Background

We publish a Docker image that includes DuckDB. The quickstart has now been changed (#6092) and therefore we can remove the DuckDB Docker image.

### Testing Details

How were the changes tested? they weren't, I'm assuming that the changes to the build files are correct <g>

### Notes

Should we also prune the [`*duckdb` images](https://hub.docker.com/r/treeverse/lakefs/tags?page=1&name=duckdb) from Docker Hub?

### CODE REVIEW

**Comment by rmoff** [comment]
Do not merge until #6092 is merged

**Review by arielshaqed — APPROVED** [review]
# :-)

**Comment by rmoff** [comment]
@arielshaqed Should we also prune the [*duckdb images](https://hub.docker.com/r/treeverse/lakefs/tags?page=1&name=duckdb) from Docker Hub? If so who has creds to do this? thanks.

**Pull request merged**
